generator/app.go

```diff
@@ -55,13 +60,93 @@ const (
 
 	regexRawGitHubURL = "https://raw.githubusercontent.com/(?P<org>[^/]+)/(?P<repo>[^/]+)/(?P<branch>[^/]+)/(?P<path>.*)"
 	regexGitHubURL    = "https://github.com/(?P<org>[^/]+)/(?P<repo>[^/]+)/(blob|tree)/(?P<branch>[^/]+)/(?P<path>.*)"
+
+	// For KEPs automation
+	alpha  = "alpha"
+	beta   = "beta"
+	stable = "stable"
+	kepURL = "https://storage.googleapis.com/k8s-keps/keps.json"
 )
 
 var (
 	baseGeneratorDir = ""
 	templateDir      = "generator"
+	releases         = Releases{}
 )
 
+// KEP represents an individual KEP holding its metadata information.
+type KEP struct {
+	Name            string `json:"name"`
+	Title           string `json:"title"`
+	KepNumber       string `json:"kepNumber"`
+	OwningSig       string `json:"owningSig"`
+	Stage           string `json:"stage"`
+	LatestMilestone string `json:"latestMilestone"`
+}
+
+type Releases struct {
+	Latest         string
+	LatestMinusOne string
+	LatestMinusTwo string
+}
+
+func getLastThreeK8sReleases() (Releases, error) {
+	ctx := context.Background()
+	client := github.NewClient(nil)
+
+	releases, _, err := client.Repositories.ListReleases(ctx, "kubernetes", "kubernetes", nil)
+	if err != nil {
+		return Releases{}, err
+	}
+	var result Releases
+	result.Latest = strings.Split(strings.TrimPrefix(*releases[0].TagName, "v"), ".")[0] + "." + strings.Split(strings.TrimPrefix(*releases[0].TagName, "v"), ".")[1]
+	result.LatestMinusOne = strings.Split(strings.TrimPrefix(*releases[1].TagName, "v"), ".")[0] + "." + strings.Split(strings.TrimPrefix(*releases[1].TagName, "v"), ".")[1]
+	result.LatestMinusTwo = strings.Split(strings.TrimPrefix(*releases[2].TagName, "v"), ".")[0] + "." + strings.Split(strings.TrimPrefix(*releases[2].TagName, "v"), ".")[1]
+	return result, nil
+}
+
+func getReleases() Releases {
+	return releases
+}
+
+func fetchAndFilterKEPs(owningSig string, releases Releases) (map[string][]KEP, error) {
+	url, err := url.Parse(kepURL)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing url: %v", err)
+	}
+
+	req, err := http.NewRequest("GET", url.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating request: %v", err)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("Error fetching KEPs: %v", err)
+	}
+	defer resp.Body.Close()
```

Comment:
nit: This will run O(number of groups) times. Although the number is very trivial, I am slightly worried about reaching GitHub rate limits. If we can fetch the KEPs and cache the result in a variable, that could be best.

Reply:
I think this is returning from GCS not GitHub, so maybe less concerned about rate limits. That said, fetching once and caching still seems like a good idea.

Reply:
implemented in the refreshed commit. Thanks for pointing.